http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -423,5 +425,5 @@ object ResourceTypeName {
 }
 
 final case object NotFoundException extends NoStackTrace
-final case class AuthProviderException(traceId: TraceId, msg: String)
-    extends LeoException(message = s"${traceId} | AuthProvider error: $msg")
+final case class AuthProviderException(traceId: TraceId, msg: String, code: StatusCode)
```

Comment:
are there cases where status code is not `StatusCodes.Unauthorized` for `AuthProviderException`? also I'm not sure which change corresponds to the title 🤔

Reply:
This is for a user liason issue. 
Sometimes SAM can return other codes, for example, for the issue I'm investigating now SAM is returning a 400. I don't want to eat the status from SAM in that case.

Reply:
Hm, why _are_ we returning the stack trace currently? Is there some logic where we return the stack trace for 500 but not other status codes?

Reply:
Yeah, that is here, but I'm also a little confused because it should match `case leoException` which does not pass the stacktrace along.  https://github.com/DataBiosphere/leonardo/blob/develop/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/api/HttpRoutes.scala#L65

Reply:
you can also just extend `NoStackTrace`, that way stacktrace is not calculated in the first place, which is more desirable if we don't care about stacktrace here.

Reply:
Yeah makes me wonder if something more subtle is going on. Like maybe we _do_ throw `AuthProviderException` here but something downstream throws a `Throwable` wrapping the original error, causing a stack trace to be returned. Just speculating though

Reply:
yeah... I modified the generic `Throwable` case in `HttpRoutes` to not pass in the stacktrace. 
This should work for all exceptions and accomplish the same thing as @Qi77Qi suggestion, but for all exceptions by default